hicommonwealth/commonwealth
# Added support for fetching trending communities from fetch communities list api

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8760

## Description of Changes
- Moved trending communities algo from frontend to communities list API.
- Renamed `thread_count` to `lifetime_thread_count` in communities table
- Added `order_by: 'last_30_day_thread_count'` filter in communities list route
- Added `include_last_30_day_thread_count` as an optional param to communities list route

## "How We Fixed It"
N/A

## Test Plan
- Visit `/dashboard/global` with logged in state
- Verify the trending communities list displays valid suggestions
   - communities which user has joined and they have new thread/comment activity after user's last session will be display at the top
   - then communities with higher 30 day thread count would be displayed
- Verify there are no regressions relation to `thread_count` -> `lifetime_thread_count` renaming


## Deployment Plan
N/A

## Other Considerations
This leads to new cleanup/refinement 
- https://github.com/hicommonwealth/commonwealth/issues/8967
- https://github.com/hicommonwealth/commonwealth/issues/8966
